akka/src/main/scala/chatapp/Chat.scala

```diff
@@ -99,123 +97,125 @@ class Chat(initiator: ActorRef[Client]) {
         }
       }
     }
-
-    client ! ClLeft(didLogout, accumulator)
+    client ! ClLeft(chat, didLogout, accumulator)
   }
 }
 
 sealed trait ClientMsg
-final case class ClBefriend(client: ActorRef[Client]) extends ClientMsg
-final case class ClLogout()                           extends ClientMsg
-final case class ClLeft(chat: ActorRef[Chat], didLogout: Boolean, accumulator: ActorRef[Accumulator]) extends ClientMsg
-final case class ClAccepted(chat: ActorRef[Chat], accumulator: ActorRef[Accumulator]) extends ClientMsg
-final case class ClForward(chat: ActorRef[Chat], payload: Array[Byte]) extends ClientMsg
-final case class ClAct(behavior: BehaviorFactory, accumulator: ActorRef[Accumulator]) extends ClientMsg
+final case class ClBefriend(client: ActorRef[ClientMsg]) extends ClientMsg
+final case class ClLogout() extends ClientMsg
+final case class ClLeft(chat: ActorRef[ChatMsg], didLogout: Boolean, accumulator: ActorRef[AccMsg]) extends ClientMsg
+final case class ClAccepted(chat: ActorRef[ChatMsg], accumulator: ActorRef[AccMsg]) extends ClientMsg
+final case class ClForward(chat: ActorRef[ChatMsg], payload: Array[Byte], accumulator: ActorRef[AccMsg]) extends ClientMsg
+final case class ClAct(behavior: BehaviorFactory, accumulator: ActorRef[AccMsg]) extends ClientMsg
 
 
-class Client(id: Long, directory: Directory, seed: Long) {
+class Client(id: Long, directory: ActorRef[DirMsg], seed: Long) {
 
-  val friends = mutable.ArrayBuffer[ActorRef[Client]]()
-  val chats   = mutable.ArrayBuffer[ActorRef[Chat]]()
+  val friends = new mutable.ArrayBuffer[ActorRef[ClientMsg]]()
+  val chats   = new mutable.ArrayBuffer[ActorRef[ChatMsg]]()
   val rand = new SimpleRand(seed)
-  val dice = new DiceRoll(rand)
```

Comment:
Why removing the `DiceRoll`?
Pony also still has it: https://github.com/sblessing/chat-app/blob/master/pony/chat.pony#L122
Would be good to make things consistent, and avoid having too many variations of things.

Reply:
see https://github.com/smarr/chat-app/pull/1#discussion_r458927685

Reply:
done

Reply:
thanks